### PR TITLE
fix: authentication issues

### PR DESCRIPTION
**Pull request by dhargitai**
## Description
This fixes the authentication issues on the production environment where the authentication cookie is set as HTTP only so the client side cannot access it.

## Related Issue
VBC-339, VBC-336, VBC-226

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/api-client/src/api/customers/logout/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+} from '../../../helpers/consts';
+
+export const logoutCustomer: Endpoints['logoutCustomer'] = (
+  context: BigcommerceIntegrationContext
+) => {
+  const {
+    config: { secureCookies },
+    res
+  } = context;
+  const cookieOptions = {
```
Comment (zoltantakacs-born) [inline packages/api-client/src/api/customers/logout/index.ts:15]: maxAge: 0 would delete the cookie

File: packages/composables/src/useUser/actions/logout.ts [hunk 2]
```diff
@@ -1,21 +1,14 @@
...
  * `logOut` method in `useUser` composable.
  * @param {Context} context An auto-generated value prepended to the method as a first parameter.
  */
 export const logOut = async (context: Context): Promise<void> => {
-  context.$bigcommerce.config.app.$cookies.remove(COOKIE_KEY_CUSTOMER_DATA);
-  context.$bigcommerce.config.app.$cookies.remove(
-    COOKIE_KEY_SHOP_SESSION_TOKEN
-  );
-  context.$bigcommerce.config.app.$cookies.remove(COOKIE_KEY_SHOPPER_PREF);
+  context.$bigcommerce.api.logoutCustomer();
```
Comment (zoltantakacs-born) [inline packages/composables/src/useUser/actions/logout.ts:9]: shouldn't we wait for success before go further? although that only makes sense with error handling

File: packages/api-client/src/api/customers/logout/index.ts (hunk 1 above)
Comment (dhargitai) [reply packages/api-client/src/api/customers/logout/index.ts:15]: Thanks!

File: packages/composables/src/useUser/actions/logout.ts (hunk 2 above)
Comment (dhargitai) [reply packages/composables/src/useUser/actions/logout.ts:9]: The endpoint method is not async so it waits for it.

Comment (zoltantakacs-born) [reply packages/composables/src/useUser/actions/logout.ts:9]: but this is an async network call from client to server, right?

**Review by scernisevs — APPROVED** [review]
LGTM! :) thank you very much David

File: packages/composables/src/useUser/actions/logout.ts (hunk 2 above)
Comment (dhargitai) [reply packages/composables/src/useUser/actions/logout.ts:9]: There's no any async operation on the endpoint, it just sets the cookie deletion headers on the response (hence in the browser) as it's interpreted. There's no need to use `await`, it already waits for the API response inside this async method.

**Review by zoltantakacs-born — APPROVED** [review]

**Pull request merged**